seirmo/figures.py

```diff
@@ -0,0 +1,99 @@
+#
+# This file is part of seirmo (https://github.com/SABS-R3-Epidemiology/seirmo/)
+# which is released under the BSD 3-clause license. See accompanying LICENSE.md
+# for copyright notice and full license details.
+#
+
+import plotly.graph_objects as go
+
+
+class IncidenceNumberPlot(object):
+    """
+
+    Creates incidence number plots for the app
+
+    """
+    def __init__(self):
+        super(IncidenceNumberPlot, self).__init__()
+
+        # Instantiate an empty plotly figure
+        self._fig = go.Figure()
+
+        """
+        Add a bar plot trace for the given incidence number data
+        in the figure
+
+        Parameters
+        ----------
+        data: pandas.DataFrame
+        time_key: docstring
+        inc_key: docstring
+        """
+    def add_data(self, data, time_key='Time',
+                 inc_key='Incidence Number'):
+
+        # Plot a bar chart for the data
+        self._fig.add_trace(
+            go.Bar(
+                x=data[time_key],
+                y=data[inc_key],
+                name='cases'
+            )
+        )
+
+        # Add axis labels; Raise error when the x and y axis labels
+        # do not match the old ones if they already exist;
+        # otherwise update the labels
+        old_x_label = self._fig['layout']['xaxis']['anchor']
+        old_y_label = self._fig['layout']['yaxis']['anchor']
+        if (old_x_label or old_y_label):
```

Comment:
I would potentially separate the x and y label checking here, i.e.
```
if (old_x_label is not None) and (old_x_label != time_key):
   warnings.warn(
              'The time key does not match the pervious time label. The x label was set to the new '
              'time key, but may now no longer be appropriate for the previously set data.')

# same for y axis here
...

# Update axes
self._fig.update_layout(
                xaxis_title=time_key,
                yaxis_title=inc_key)
```
and then the same for the y axis. That makes it easier to understand which labels may now be wrong.